### ISSUE TITLE
TypedArray.set() asserts if the first parameter is undefined.

### ISSUE THREAD

**Issue by harold-b** [svaarala/duktape#1285]
Example:
``` javascript

var a = new Float32Array(2);
a.set( undefined ); // Causes assert crash in builds with DUK_USE_ASSERTIONS
```
Maybe I'm mistaken, but isn't the expected behavior here to throw if the parameter is undefined even in debug builds?

This is called from the protected variant `duk_pcall_method()`. See the offending call stack:

```
duk_default_fatal_handler(void * udata, const char * msg) Line 10537    C
duk_known_hobject(duk_hthread * ctx, int idx) Line 18310    C
duk_bi_typedarray_set(duk_hthread * ctx) Line 25449 C
duk__handle_call_inner(duk_hthread * thr, int num_stack_args, unsigned int call_flags, int idx_func) Line 59505 C
duk_handle_call_unprotected(duk_hthread * thr, int num_stack_args, unsigned int call_flags) Line 59038  C
duk__js_execute_bytecode_inner(duk_hthread * entry_thread, unsigned int entry_callstack_top) Line 73087 C
duk_js_execute_bytecode(duk_hthread * exec_thr) Line 70899  C
duk__handle_call_inner(duk_hthread * thr, int num_stack_args, unsigned int call_flags, int idx_func) Line 59453 C
duk_handle_call_protected(duk_hthread * thr, int num_stack_args, unsigned int call_flags) Line 58936    C
duk_pcall_method(duk_hthread * ctx, int nargs) Line 13382   C
Musashi::ScriptGameSys::InitScriptApp() Line 424    C++
```
I am running off commit f7e0d6f72195507382a4460e7f085b17e7280b1a from the #1264 PR.

**Comment by fatcerberus** [comment]
That's definitely a bug.  Node.js:
```
> var a = new Float32Array(2)
undefined
> a.set(undefined)
TypeError: Cannot read property 'length' of undefined
    at Float32Array.set (native)
    at repl:1:3
    at realRunInThisContextScript (vm.js:22:35)
    at sigintHandlersWrap (vm.js:98:12)
    at ContextifyScript.Script.runInThisContext (vm.js:24:12)
    at REPLServer.defaultEval (repl.js:346:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.onLine (repl.js:545:10)
    at emitOne (events.js:101:20)
```

**Comment by harold-b** [comment]
I had tested on chrome console and got a throw as well:

``` javascript
new Float32Array(2).set( undefined )
VM182:1 Uncaught TypeError: Cannot read property 'length' of undefined
    at Float32Array.set (native)
    at <anonymous>:1:21
(anonymous) @ VM182:1
```

**Comment by svaarala** [comment]
Thanks @harold-b! It's indeed a bug present in 1.x but is present in 2.0.0 so I'll fix it into 2.0.1 too.
